### PR TITLE
Update java-rest-service to Spring Boot 3.3.0

### PR DESCRIPTION
**Pull request by trisberg**

### CODE REVIEW

File: fragments/testcontainers-mysql/accelerator.yaml [hunk 1]
```diff
@@ -3,29 +3,6 @@ accelerator:
 
 engine:
   merge:
-
```
[the remark this answers is not in the document]
Comment (ericbottard) [reply fragments/testcontainers-mysql/accelerator.yaml:6]: I see there are TextProcessor directives in the affected files now, but the triggering (and thus responsibility) of TextProcessor is not in this fragment anymore. Is this really correct?

File: java-rest-service/pom.xml [hunk 2]
```diff
@@ -46,21 +46,75 @@
...
+        <!-- #ENDIF -->
+
         <!-- Infrastructure -->
+        <!-- #IF(#databaseMigrationTool == 'flyway') -->
         <dependency>
             <groupId>org.flywaydb</groupId>
             <artifactId>flyway-core</artifactId>
         </dependency>
+        <!-- #ENDIF -->
+        <!-- #IF(#databaseMigrationTool == 'flyway' && #databaseType == 'postgres') -->
```
Comment (ericbottard) [inline java-rest-service/pom.xml:86]: We _could_ remove the first condition and extend the check for flyway that starts at line 80 until the end, using nested directives

File: java-rest-service/README.md [hunk 3]
```diff
@@ -47,33 +47,34 @@ You will need a local database running, see [DATABASE.md](DATABASE.md#local).
 
 After that it is a good habit to compile the test classes and execute those tests to see if your application is still behaving as you would expect:
 
---- StartMaven
+#IF(#buildTool == 'maven')
```
[the remark this answers is not in the document]
Comment (trisberg) [reply java-rest-service/README.md:50]: I've not seen this being a problem in any parser - do you have an example?

This looks fine: https://gist.github.com/trisberg/60a24cd5c4209e1d3a0b3ca7838953ba

File: fragments/testcontainers-mysql/accelerator.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (trisberg) [reply fragments/testcontainers-mysql/accelerator.yaml:6]: The dependecies got added here but I found adding them to the top of the dependencies looked ugly and the whole approach seemed very brittle. So, I moved all potential dependencies to the `pom.xml` / `build.gradle.kts` with TextPreprocessor directives. I don't think these fragments are really reusable anyway unless we have a way of cleanly inject dependencies.

File: java-rest-service/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ericbottard) [reply java-rest-service/README.md:50]: If you're ok with seeing the directives appear in the preview, yes.
I guess it boils down to preference.
Consider this too: if we had made the directives be `# IF(...)` (with a space) and thus creating an accidental Level1 header, would you leave them like that or wrap them in a comment structure (however broken it is)?

File: fragments/testcontainers-mysql/accelerator.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ericbottard) [reply fragments/testcontainers-mysql/accelerator.yaml:6]: This needs to be addressed IMO, and would improve readability even further

**Comment by ericbottard** [comment]
Decided to go with #497 for the time being

File: java-rest-service/pom.xml (hunk 2 above)
Comment (trisberg) [reply java-rest-service/pom.xml:86]: @ericbottard  I tried to nest the `#IF` directives but couldn't get it to work - do we have this documented, I'm not exactly sure what to expect.

I tried something like this but that did not fully work
```
#IF(#tests)
This is nested test:
#IF(#javaVersion == '17')
Java 17
#ENDIF
#IF(#javaVersion == '21')
Java 21
#ENDIF
#ENDIF
```
I got the java version shown with tests set to false

This worked as I would expect:
```
#IF(#tests)
This is explicit test:
#IF(#tests && #javaVersion == '17')
Java 17
#ENDIF
#IF(#tests && #javaVersion == '21')
Java 21
#ENDIF
#ENDIF
```

Comment (ericbottard) [reply java-rest-service/pom.xml:86]: Created https://github.com/pivotal/acc-engine/issues/481

**Pull request closed without merging**